src/core/manager.c

```diff
@@ -3834,6 +3834,14 @@ static int manager_run_generators(Manager *m) {
         if (r == 0) {
                 r = manager_execute_generators(m, paths, /* remount_ro= */ true);
                 _exit(r >= 0 ? EXIT_SUCCESS : EXIT_FAILURE);
+
+        } else if (r == -EPERM) {
```

Comment:
maybe use ERRNO_IS_PRIVILEGE()?

Reply:
hmm, we use FORK_LOG in the safe_fork() invocation, so there will already be a loud message about lack of privs being printed here if we lacked the privs.

Maybe move this into safe_fork() instead, i.e. add a new flag `FORK_MOUNTNS_GRACEFUL` which will implicitly cancel the effect of `FORK_NEW_MOUNTNS` and `FORK_PRIVATE_TMP` flag if it can't acquire a new namespace in the child. It would still fork off the child, and it would print a debug message only in case it does the fallback thing.

Reply:
As I wrote on an earlier version of the patch, I think the message should be at debug level. We support running without namespaces, so we shouldn't warn too loudly.